tests/python_package_test/test_engine.py

```diff
@@ -1067,6 +1067,29 @@ def test_early_stopping_min_delta(first_only, single_metric, greater_is_better):
         assert np.greater_equal(last_score, best_score - min_delta).any()
 
 
+@pytest.mark.parametrize("early_stopping_min_delta", [1e3, 0.0])
+def test_early_stopping_min_delta_via_global_params(early_stopping_min_delta):
+    X, y = load_breast_cancer(return_X_y=True)
+    num_trees = 5
+    params = {
+        "num_trees": num_trees,
```

Comment:
```suggestion
        "num_trees": num_trees,
        "num_leaves": 5,
```

Would you consider adding a small constraint on `num_leaves`, to make this even smaller and faster?